test/jdk/java/awt/ScrollPane/ScrollPositionTest.java

```diff
@@ -0,0 +1,104 @@
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4008152
+ * @summary ScrollPane position does not return correct values
+ * @key headful
+ * @run main ScrollPositionTest
+ */
+
+import java.awt.Adjustable;
+import java.awt.BorderLayout;
+import java.awt.Canvas;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.ScrollPane;
+import java.awt.event.AdjustmentEvent;
+import java.awt.event.AdjustmentListener;
+
+public class ScrollPositionTest {
+    static Frame frame;
+    public static Object lock = new Object();
+    static int i = 0;
+    static Point p;
+    static ScrollPane sp;
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+        try {
+            EventQueue.invokeAndWait(() -> {
+                frame = new Frame("Scroll Position Test");
+                frame.setLayout(new BorderLayout());
+                frame.setSize(200, 200);
+                sp = new ScrollPane(ScrollPane.SCROLLBARS_AS_NEEDED);
+                Canvas canvas = new Canvas();
+                canvas.setSize(300, 300);
+                sp.add(canvas);
+                frame.add("Center", sp);
+                frame.setLocationRelativeTo(null);
+                frame.setVisible(true);
+            });
+            robot.waitForIdle();
+            robot.delay(1000);
+            EventQueue.invokeAndWait(() -> {
+                Adjustable saH = sp.getHAdjustable();
+                saH.addAdjustmentListener(new TestAdjustmentListener());
+            });
+            for (i = 0; i < 1000; i++) {
+                EventQueue.invokeAndWait(() -> {
+                    p = new Point(i % 100, i % 100);
+                    sp.setScrollPosition(p);
+                });
+
+                robot.waitForIdle();
+                robot.delay(10);
+                EventQueue.invokeAndWait(() -> {
+                    if (!sp.getScrollPosition().equals(p)) {
+                        throw new RuntimeException("Test failed. " + i + " : " +
+                                "Expected " + p + ", but Returned: " + sp.getScrollPosition());
+                    }
+                });
+            }
+            System.out.println("Test Passed.");
+        } finally {
+            EventQueue.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
+    }
+
+    private static class TestAdjustmentListener implements AdjustmentListener {
+        public void adjustmentValueChanged(AdjustmentEvent e) {
+            System.out.println("AdjEvent caught:" + e);
+            synchronized (ScrollPositionTest.lock) {
+                ScrollPositionTest.lock.notify();
+            }
```

Comment:
Better to use CountDownLatch than wait-notify but there is no lock.wait(timeout) so not sure whom it is notifying so it seems to be a no-op

Reply:
Then shall I remove it ?

Reply:
Anyhow I've removed it and seems like no-op.